ontology/obo/tree.go

```diff
@@ -0,0 +1,160 @@
+package obo
+
+import (
+	"fmt"
+	"github.com/vertgenlab/gonomics/exception"
+	"github.com/vertgenlab/gonomics/fileio"
+	"log"
+)
+
+// BuildTree takes a slice of Obo structs and builds a tree, setting
+// the Parents and Children fields to point to related nodes.
+func BuildTree(terms []Obo) map[string]*Obo {
+	termMap := makeTermMap(terms)
+
+	var parentId string
+	var foundInMap bool
+	var parentTerm *Obo
+	var isAItem IsADescription
+
+	// Now we iterate through terms and set up the tree
+	for i := range terms {
+		term := &terms[i] //get a memory copy
+		for _, isAItem = range term.IsA {
+			parentId = isAItem.ParentId
+			if parentTerm, foundInMap = termMap[parentId]; foundInMap {
+				term.Parents = append(term.Parents, parentTerm)
+				parentTerm.Children = append(parentTerm.Children, term)
+			} else {
+				//log.Fatalf("Error: The term with ID \"%s\" has a parent with ID \"%s\" that "+
+				//	"is not found in the Obo file.\n", term.Id, parentId)
+			}
+		}
+	}
+	return termMap
+}
+
+// termToDot is a helper function of ToDot that converts and individual Obo struct to a
+// DOT format digraph descriptor, and recursively does so for all children of the input Obo struct.
+func termToDot(term Obo, out *fileio.EasyWriter, visitedNode map[string]bool) {
```

Comment:
We have some complexity in the code (e.g. visitedNode map) because we don't know the root.  I think this should probably take a pointer to the root instead of a term and visitedNode